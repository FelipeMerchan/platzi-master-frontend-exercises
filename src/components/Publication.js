import React from 'react'

import './styles/Publication.css'

function Publication(props) {
  return (
    <article className="Publication">
      <div className="Publication__data">
        <span>
          El {props.date} <br />
          Por <span className="text-red">{props.autor}</span><br />
          <small>{props.comments} comentarios</small>
        </span>
      </div>
      <div className="Publication__details">
        <h2 className="Publication__title mb">
          {props.title}
        </h2>
        <p className="Publication__text mb-2">
          {props.summary}
        </p>
        <a href={props.url} className="Publication__link">Leer m√°s</a>
      </div>
    </article>
  )
}

export default Publication