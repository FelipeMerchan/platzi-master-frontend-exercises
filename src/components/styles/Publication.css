.Publication {
  display: flex;
  justify-content: space-between;
  padding: var(--space-2) 0;
  border-bottom: 1px solid var(--redColor);
  font-family: var(--font-2);
  color: var(--typographyColor);
}

.Publication:last-child {
  border: none;
}

.Publication__data {
  width: 20%;
  margin-right: 1%;
}

.Publication__author {
  color: #c84451;
}

.Publication__details {
  width: 79%;
}

.Publication__title {
  color: #c84451;
  ;
}

.Publication__text {
  line-height: 1.2;
}

.Publication__link {
  color: #c84451;
  ;
}

@media screen and (max-width: 768px) {
  .Publication {
    display: block;
  }

  .Publication__data,
  .Publication__details {
    width: auto;
  }

  .Publication__data {
    margin-bottom: var(--space);
  }
}