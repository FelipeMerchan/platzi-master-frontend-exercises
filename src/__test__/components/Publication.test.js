import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import Publication from '../../components/Publication'

describe('Probar el componente <Publication />', () => {
  const publication = shallow(<Publication />)

  test('Render del componente Publication', () => {
    expect(publication.length).toEqual(1)
  })
  test('Render de los 2 hijos', () => {
    expect(publication.children()).toHaveLength(2);
  });
  test('Render con el className Publication__link', () => {
    expect(publication.exists('.Publication__link')).toBeTruthy()
  })
})

describe('Publication Snapshot', () => {
  test('Comprobar la UI del componente Publication', () => {
    const publication = create(<Publication />)
    expect(publication.toJSON()).toMatchSnapshot()
  })
})