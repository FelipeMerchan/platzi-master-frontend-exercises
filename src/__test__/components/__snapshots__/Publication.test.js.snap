// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publication Snapshot Comprobar la UI del componente Publication 1`] = `
<article
  className="Publication"
>
  <div
    className="Publication__data"
  >
    <span>
      El 
       
      <br />
      Por 
      <span
        className="Publication__author"
      />
      <br />
      <small>
         comentarios
      </small>
    </span>
  </div>
  <div
    className="Publication__details"
  >
    <h2
      className="Publication__title mb"
    />
    <p
      className="Publication__text mb-2"
    />
    <a
      className="Publication__link"
    >
      Leer m√°s
    </a>
  </div>
</article>
`;
